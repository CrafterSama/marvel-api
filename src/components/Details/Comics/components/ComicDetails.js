import React from 'react';
import PropTypes from 'prop-types';
import {
  NO_DESCRIPTION,
} from '../../../../constants';

import {
  ComicDetail,
  ComicImageBox,
  ComicInfo,
  ComicCreators,
  DetailsItem,
  TextStrong,
} from '../../../../styles';

const ComicDetails = ({ data }) => {

  const publicDate = data && data.dates.filter((o) => o.type === 'onsaleDate')

  const creators = data && data.creators.items;
  const writers = data && creators.filter(c => c.role === 'writer');
  const colorists = data && creators.filter(c => c.role === 'colorist');
  const editors = data && creators.filter(c => c.role === 'editor');
  const artists = data && creators.filter(c => c.role === 'artist');
  const pencilers = data && creators.filter(c => c.role === 'penciler');
  const inkers = data && creators.filter(c => c.role === 'inker');
  const letterers = data && creators.filter(c => c.role === 'letterer');
  const covers = data && creators.filter(c => {
    return (c.role === 'penciler (cover)' || c.role === 'colorist (cover)')
    });

  return (
    <ComicDetail>
      <ComicImageBox>
        <img
          src={`${data && data.thumbnail.path}.${data && data.thumbnail.extension}`}
          alt={data && data.title}
        />
      </ComicImageBox>
      <ComicInfo>
        <h2>{data && data.title}</h2>
        <ComicCreators>
          <DetailsItem>
            <TextStrong>Public Time: </TextStrong>
            {data && publicDate.map((d, key) => (
              <span key={key}>{d.date}</span>
            ))}
          </DetailsItem>
          {data && data.creators && (
            <>
              {writers &&
                <DetailsItem>
                  <>
                    <TextStrong>Writers: </TextStrong>
                  {data && writers.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {pencilers &&
                <DetailsItem>
                  <>
                    <TextStrong>Pencilers: </TextStrong>
                  {data && pencilers.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {artists &&
                <DetailsItem>
                  <>
                    <TextStrong>Artists: </TextStrong>
                  {data && artists.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {colorists &&
                <DetailsItem>
                  <>
                    <TextStrong>Colorists: </TextStrong>
                  {data && colorists.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {editors &&
                <DetailsItem>
                  <>
                    <TextStrong>Editors: </TextStrong>
                  {data && editors.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {inkers &&
                <DetailsItem>
                  <>
                    <TextStrong>Inkers: </TextStrong>
                  {data && inkers.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {letterers &&
                <DetailsItem>
                  <>
                    <TextStrong>Letterers: </TextStrong>
                  {data && letterers.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
              {covers &&
                <DetailsItem>
                  <>
                    <TextStrong>Cover Artists: </TextStrong>
                  {data && covers.map((w, key) => (
                      <span key={key}>{w.name}</span>
                    ))}
                  </>
                </DetailsItem>
              }
            </>
          )}
        </ComicCreators>
        <DetailsItem>
          {data && data.description
            ? data.description
            : NO_DESCRIPTION
          }
        </DetailsItem>
      </ComicInfo>
    </ComicDetail>
  );
};

ComicDetails.propTypes = {
  data: PropTypes.object.isRequired
};

export default ComicDetails;
