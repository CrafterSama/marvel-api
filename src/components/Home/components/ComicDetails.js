import React from 'react'

const ComicsDetails = () => {
  return (
    <div>
      Character
    </div>
  )
}

export default ComicsDetails
