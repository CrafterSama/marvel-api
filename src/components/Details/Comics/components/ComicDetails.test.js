import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import ComicDetails from './ComicDetails';

configure({ adapter: new Adapter() });

describe('<ComicDetails />', () => {

  const data = {
    title:
      'New Mutants by Zeb Wells: The Complete Collection (Trade Paperback)',
    dates: [],
    creators: { items: [] },
    description: 'description',
    thumbnail: {
      path: 'http://i.annihil.us/u/prod/marvel/i/mg/f/e0/5bb5301610695',
      extension: 'jpg',
    },
  };

  test('matches snapshot', () => {
    const wrapper = shallow(
      <ComicDetails data={data} />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
