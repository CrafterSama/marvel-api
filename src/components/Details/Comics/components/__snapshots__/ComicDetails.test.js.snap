// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicDetails /> matches snapshot 1`] = `
<styles__ComicDetail>
  <styles__ComicImageBox>
    <img
      alt="New Mutants by Zeb Wells: The Complete Collection (Trade Paperback)"
      src="http://i.annihil.us/u/prod/marvel/i/mg/f/e0/5bb5301610695.jpg"
    />
  </styles__ComicImageBox>
  <styles__ComicInfo>
    <h2>
      New Mutants by Zeb Wells: The Complete Collection (Trade Paperback)
    </h2>
    <styles__ComicCreators>
      <styles__DetailsItem>
        <styles__TextStrong>
          Public Time: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Writers: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Pencilers: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Artists: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Colorists: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Editors: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Inkers: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Letterers: 
        </styles__TextStrong>
      </styles__DetailsItem>
      <styles__DetailsItem>
        <styles__TextStrong>
          Cover Artists: 
        </styles__TextStrong>
      </styles__DetailsItem>
    </styles__ComicCreators>
    <styles__DetailsItem>
      description
    </styles__DetailsItem>
  </styles__ComicInfo>
</styles__ComicDetail>
`;
